mlrun/api/main.py

```diff
@@ -224,6 +226,105 @@ async def move_api_to_online():
+        runs = await fastapi.concurrency.run_in_threadpool(
+            get_db().list_distinct_runs_uids,
+            db_session,
+            project="*",
+            requested_logs=False,
+            only_uids=False,
+        )
+        # each result contains either run_uid or None
+        # if it's None it means something went wrong and we should skip it
+        # if it's run_uid it means we requested logs collection for it and we should update it's requested_logs field
+        results = await asyncio.gather(*[_start_log_for_run(run) for run in runs])
+        runs_to_update_requested_logs = [result for result in results if result]
+
+        if len(runs_to_update_requested_logs) > 0:
+            # update the runs to indicate that we have requested log collection for them
+            await fastapi.concurrency.run_in_threadpool(
+                get_db().update_runs_requested_logs,
+                db_session,
+                uids=runs_to_update_requested_logs,
+            )
+
+    finally:
+        close_session(db_session)
+
+
+async def _start_log_for_run(run: dict) -> typing.Union[str, None]:
+    """
+    Starts log collection for a specific run
+    :param run: run object
+    :return: the run_uid of the run if log collection was started, None otherwise
+    """
+    # using semaphore to limit the number of concurrent log collection requests
+    # this is to prevent opening too many connections to many connections
+    async with start_logs_limit:
+        logs_collector_client = (
+            mlrun.api.utils.clients.log_collector.LogCollectorClient()
+        )
+        run_kind = run.get("metadata", {}).get("labels", {}).get("kind", None)
+        project_name = run.get("metadata", {}).get("project", None)
+        run_uid = run.get("metadata", {}).get("uid", None)
+        # empty kind means it's a local run, the log collector doesn't support it
+        if not run_kind:
+            # we mark the run as requested logs collection so we won't iterate over it again
+            return run_uid
+        try:
+            runtime_handler = get_runtime_handler(run_kind)
+            label_selector = runtime_handler.resolve_label_selector(
+                project=project_name,
+                object_id=run_uid,
+                class_mode=RuntimeClassMode.run,
+            )
+            success, _ = await logs_collector_client.start_logs(
+                run_uid=run_uid,
+                selector=label_selector,
+                project=project_name,
+                raise_on_error=True,
+            )
+            if success:
+                # update the run to mark that we requested logs collection for it
+                return run_uid
+
+        except Exception as exc:
+            logger.warning(
+                "Failed to start logs for run",
+                run_uid=run_uid,
+                exc=exc,
+                traceback=traceback.format_exc(),
+            )
+            return None
```

Comment:
dangerous pattern to ignore the exception here. why not letting the callee handle it?

Reply:
the logic is that the ones that succeeded will return their uid and the ones that failed will return None, that way the results will be easy to process. 

I can add an option of `raise_error` which will by default will be true and in our logic it will be False

Reply:
added `raise_error` option